src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/ContaminationRecord.java

```diff
@@ -0,0 +1,107 @@
+        this.level = level;
+        this.contamination = contamination;
+    }
+
+    public String getLevel() {
+        return level;
+    }
+
+    public double getContamination() {
+        return contamination;
+    }
+
+    //----- The following two public static methods read and write contamination files
+    public static void writeContaminationTable(final List<ContaminationRecord> records, final File outputTable) {
+        try ( ContaminationRecord.ContaminationTableWriter writer = new ContaminationRecord.ContaminationTableWriter(outputTable) ) {
+            writer.writeAllRecords(records);
+        } catch (IOException e){
+            throw new UserException(String.format("Encountered an IO exception while writing to %s.", outputTable));
+        }
+    }
+
+    public static List<ContaminationRecord> readContaminationTable(final File tableFile) {
+        try( ContaminationTableReader reader = new ContaminationTableReader(tableFile) ) {
+            return reader.toList();
+        } catch (IOException e){
+            throw new UserException(String.format("Encountered an IO exception while reading from %s.", tableFile));
+        }
+    }
+
+    //-------- The following methods are boilerplate for reading and writing contamination tables
+    private static class ContaminationTableWriter extends TableWriter<ContaminationRecord> {
+        private ContaminationTableWriter(final File output) throws IOException {
+            super(output, ContaminationTableColumn.COLUMNS);
+        }
+
+        @Override
+        protected void composeLine(final ContaminationRecord record, final DataLine dataLine) {
+            dataLine.set(ContaminationTableColumn.LEVEL.toString(), record.getLevel())
+                    .set(ContaminationTableColumn.CONTAMINATION.toString(), record.getContamination());
+        }
+    }
+
+    private static class ContaminationTableReader extends TableReader<ContaminationRecord> {
+        public ContaminationTableReader(final File file) throws IOException {
+            super(file);
+        }
+
+        @Override
+        protected ContaminationRecord createRecord(final DataLine dataLine) {
+            final String sample = dataLine.get(ContaminationTableColumn.LEVEL);
+            final double contamination = dataLine.getDouble(ContaminationTableColumn.CONTAMINATION);
+            return new ContaminationRecord(sample, contamination);
+        }
+    }
+
+    private enum ContaminationTableColumn {
+        LEVEL("level"),
+        CONTAMINATION("contamination");
+
+        private final String columnName;
+
+        ContaminationTableColumn(final String columnName) {
+            this.columnName = Utils.nonNull(columnName);
+        }
+
+        @Override
+        public String toString() {
+            return columnName;
+        }
+
+        public static final TableColumnCollection COLUMNS = new TableColumnCollection((Object[]) values());
```

Comment:
It turns out you do if you want to call `values()` -- not an error, but Travis will fail because it yields a compiler warning -- but I don't really care about that and so `new TableColumnCollection(LEVEL, CONTAMINATION)` avoids the cast.